java/core/src/main/java/com/whylogs/core/views/DatasetProfileView.java

```diff
@@ -0,0 +1,59 @@
+package com.whylogs.core.views;
+
+import java.util.*;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.ToString;
+
+// TODO: extend writable when we do Protobuf
+@AllArgsConstructor
+@Getter
+@ToString
+public class DatasetProfileView {
+  private HashMap<String, ColumnProfileView> columns;
+  private Date datasetTimestamp;
```

Comment:
I assume we don't want to take on any dependencies that we don't absolutely need (like guava). Since these things are supposed to be immutable, did we consider at least using things like `Collections.unmodifiableMap()` to make sure we aren't shooting ourselves in the foot by violating that anywhere?

Reply:
That's a great idea in this place. I did know there was an an unmodifiableMap. I'll look into implementing that here and in the ColumnProfileView